yt/data_objects/construction_data_containers.py

```diff
@@ -816,6 +819,11 @@ def shape(self):
 
     def _setup_data_source(self):
         self._data_source = self.ds.region(self.center, self.left_edge, self.right_edge)
+        if self._extra_data_source is not None:
```

Comment:
OK, I hadn't thought about it, but great catch.

Reply:
out of date as well -- current logic is similar but checks `self._data_source` instead